### PR TITLE
First draft of wiki notes for Lesson 7

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: Class-7-(Interfaces).md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Class 7: Interfaces and Polymorphism
+
+## In the prep work this week, the students learned
+1. What polymorphism is.
+1. What interfaces are and the difference between interfaces and abstract classes.
+1. Different interfaces that are a part of Java and how they can be used in our code.
```
Comment (jimflores5) [inline Class-7-(Interfaces).md]: Comma addition: "...Java, and how..."

File: Class-7-(Interfaces).md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+1. Best practices around unit tests and interfaces.
+
+## Announcements
+1. Check with your course manager for any important announcements.
```
Comment (jimflores5) [inline Class-7-(Interfaces).md:11]: How about a reminder that assignment 2 is due before the next class?

File: Class-7-(Interfaces).md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## Large Group Time (Instructor)
+
+### Class 7 Topics That Require Careful Attention
+1. Polymorphism has a deceptively simple definition, but can be difficult to grasp. Have students share what polymorphism is to them and share examples to drive home this particular pillar of OOP.
```
Comment (jimflores5) [inline Class-7-(Interfaces).md:16]: With any highly abstract concept, the more support we offer to the teachers, the better.  I'd add one or two clear, concrete examples here that show polymorphism - maybe one for an object and one for a method?

File: Class-7-(Interfaces).md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+1. Interfaces may not seem to serve much of a purpose to some. Go over some sample code or class diagram and explain why those behaviors were assigned to an interface. To drive it home, share what behaviors might go in an abstract class.
```
Comment (jimflores5) [inline Class-7-(Interfaces).md:18]: For instructors who may not be well versed in teaching interfaces, I'd add at least one clear, concrete example here.

File: Class-7-(Interfaces).md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+## Small Group Time: Class 6 Studio (TF Notes)
```
Comment (jimflores5) [inline Class-7-(Interfaces).md]: TA Notes!!!!  :-)

File: Class-7-(Interfaces).md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+1. This studio requires use of given interfaces in Java. Encourage students who are confused to checkout the documentation linked in the book for further explanation of how these things work!
```
Comment (jimflores5) [inline Class-7-(Interfaces).md]: Remind the TAs to check in with each of their students.

Also, point out particular areas in the studio that are likely to present problems and/or common mistakes that students could make (e.g. forgetting in import a library, extending the interface into the wrong class, etc.).

File: Class-7-(Interfaces).md (hunk 1 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md]: Done! Thanks!

File: Class-7-(Interfaces).md (hunk 2 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md:11]: Done! Thanks!

File: Class-7-(Interfaces).md (hunk 3 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md:16]: I added some direction on examples to refer to from the book.

File: Class-7-(Interfaces).md (hunk 4 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md:18]: I added some direction on providing an example from their work life.

File: Class-7-(Interfaces).md (hunk 5 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md]: Changed! Thanks!

File: Class-7-(Interfaces).md (hunk 6 above)
Comment (gildedgardenia) [reply Class-7-(Interfaces).md]: Done! Thanks!

File: Class-7-(Interfaces).md (hunk 3 above)
Comment (jimflores5) [reply Class-7-(Interfaces).md:16]: OK.

File: Class-7-(Interfaces).md (hunk 4 above)
Comment (jimflores5) [reply Class-7-(Interfaces).md:18]: Good.

**Review by jimflores5 — APPROVED** [review]

**Pull request merged**